src/confluent_kafka/src/Admin.c

```diff
@@ -4357,7 +4458,9 @@ static PyObject *Admin_c_ListOffsetsResultInfos_to_py (const rd_kafka_ListOffset
                         if (value == NULL)
                                 goto raise;
                 }
-                PyDict_SetItem(result, c_part_to_py(c_topic_partition), value);
+                key = c_part_to_py(c_topic_partition);
+                PyDict_SetItem(result, key, value);
```

Comment:
Add error handling to SetItem
```suggestion
                if (PyDict_SetItem(result, key, value) == -1) {
                        Py_DECREF(key);
                        Py_DECREF(value);
                        goto raise;
                }
```

Reply:
In `Metadata.c` there's `c_brokers_to_py` that has the same problem, you can add
`Py_DECREF(key);` before `Py_DECREF(broker);`